security/rbac/engine/engine.go

```diff
@@ -0,0 +1,207 @@
+// empty as the decision was made after all conditions evaluated to false.
+// If decision is unknown, policyNames will contain the list of policies that
+// evaluated to unknown.
+type AuthorizationDecision struct {
+	decision    Decision
+	policyNames []string
+}
+
+// Converts an expression to a parsed expression, with SourceInfo nil.
+func exprToParsedExpr(condition *expr.Expr) *expr.ParsedExpr {
+	return &expr.ParsedExpr{Expr: condition}
+}
+
+// Converts an expression to a CEL program.
+func exprToProgram(condition *expr.Expr) cel.Program {
+	// TODO(@ezou): implement the conversion from expr to CEL program.
+	return nil
+}
+
+// policyEngine is the struct for an engine created from one RBAC proto.
+type policyEngine struct {
+	action   pb.RBAC_Action
+	programs map[string]cel.Program
+}
+
+// Creates a new policyEngine from an RBAC policy proto.
+func newPolicyEngine(rbac *pb.RBAC) *policyEngine {
+	action := rbac.Action
+	programs := make(map[string]cel.Program)
+	for policyName, policy := range rbac.Policies {
+		programs[policyName] = exprToProgram(policy.Condition)
+	}
+	return &policyEngine{action, programs}
+}
+
+// Returns the decision of an engine based on whether or not AuthorizationArgs is a match,
+// i.e. if engine's action is ALLOW and match is true, we will return DecisionAllow;
+// if engine's action is ALLOW and match is false, we will return DecisionDeny.
+func getDecision(engine *policyEngine, match bool) Decision {
+	if engine.action == pb.RBAC_ALLOW && match || engine.action == pb.RBAC_DENY && !match {
+		return DecisionAllow
+	}
+	return DecisionDeny
+}
+
+// Returns the authorization decision of a single policy engine based on activation.
+// If any policy matches, the decision matches the engine's action, and the first
+//  matching policy name will be returned.
+// Else if any policy is missing attributes, the decision is unknown, and the list of
+//  policy names that can't be evaluated due to missing attributes will be returned.
+// Else, the decision is the opposite of the engine's action, i.e. an ALLOW engine
+//  will return DecisionDeny, and vice versa.
+func (engine *policyEngine) evaluate(activation interpreter.Activation) (Decision, []string) {
+	unknownPolicyNames := []string{}
+	for policyName, program := range engine.programs {
+		// Evaluate program against activation.
+		var match bool
+		out, _, err := program.Eval(activation)
+		if out == nil {
+			// Unsuccessful evaluation, typically the result of a series of incompatible
+			// `EnvOption` or `ProgramOption` values used in the creation of the evaluation
+			// environment or executable program.
+			grpclog.Warning("Unsuccessful evaluation encountered during AuthorizationEngine.Evaluate: %s", err.Error())
+			match = false
+		} else if err != nil {
+			// Successful evaluation to an error result, i.e. missing attributes.
+			match = false
+		} else {
+			// Successful evaluation to a non-error result.
+			match = out.Value().(bool)
```

Comment:
It would be better if you do a check before converting it to bool.
You can do 
`match, ok = out.Value().(bool)`
or even better, it seems CEL has some utility library that can speculate the type of out?